src/rhsm/utils.py

```diff
@@ -280,6 +280,9 @@ def fix_no_proxy() -> None:
 
 def suppress_output(func: Callable) -> Callable:
     def wrapper(*args, **kwargs):
+        devnull: TextIO = None
+        stdout: TextIO = None
+        stderr: TextIO = None
         try:
             devnull = open(os.devnull, "w")
```

Comment:
I don't think I understand what you mean regarding moving those lines. 
Would it instead be better to check if devnull is not None before assigning it to sys.stdout/sys.stderr in case open() raises an exception?
(like this: https://github.com/candlepin/subscription-manager/pull/3168/commits/96d7e0fc277fe887a3c9648edb6c705834e779af)

Reply:
It is actually not necessary, because the finally block contains `if stdout is not None` check before restoring `sys.stdout`, etc. I probably overlooked this.

Reply:
Thanks for the clarification